packages/@react-spectrum/list/src/ListView.tsx

```diff
@@ -242,51 +240,17 @@ function ListView<T extends object>(props: SpectrumListViewProps<T>, ref: DOMRef
               )
             }
             layout={layout}
-            collection={collection}
-            transitionDuration={isLoading ? 160 : 220}>
-            {(type, item) => {
+            layoutOptions={useMemo(() => ({isLoading}), [isLoading])}
+            collection={collection}>
+            {useCallback((type, item) => {
               if (type === 'item') {
-                return (
-                  <>
-                    {isListDroppable && collection.getKeyBefore(item.key) == null &&
```

Comment:
Splitting lots of this out into separate components. The callback passed to virtualizer is memoized, so we can't rely on the values accessed from closure scope being correct here. Moving it into separate components and accessing what we need via context fixes this.